### PR TITLE
Clean up Socket.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
@davidsh @stephentoub PTAL

**Comment by davidsh** [comment]
LGTM

File: src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs [hunk 1]
```diff
@@ -189,14 +194,14 @@ internal static void SelectFileDescriptor(IList socketList, IntPtr[] fileDescrip
...
-            // if a acceptSocket isn't specified, then we need to create it.
+            // If an acceptSocket isn't specified, then we need to create one.
             if (acceptSocket == null)
             {
                 acceptSocket = new Socket(_addressFamily, _socketType, _protocolType);
             }
             else
             {
-                if (acceptSocket.m_RightEndPoint != null && (!checkDisconnected || !acceptSocket._isDisconnected))
+                if (acceptSocket._rightEndPoint != null && (!checkDisconnected || !acceptSocket._isDisconnected))
```
Comment (CIPop) [inline src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs:204]: Same comment as in the previous PR: ._ doesn't look good.
We should refactor internals using caps instead (similar to `SocketAddress.Buffer`).

**Comment by CIPop** [comment]
Multiple points where ._ has been used. We should be consistent and rename internals similar to public properties instead of the _ prefix.

LGTM otherwise.

**Pull request merged**